components-api/src/main/scala/pl/touk/nussknacker/engine/api/lazyparam/LazyParameterWithPotentiallyPostponedEvaluator.scala

```diff
@@ -0,0 +1,77 @@
+package pl.touk.nussknacker.engine.api.lazyparam
+
+import pl.touk.nussknacker.engine.api.typed.typing.{Typed, TypingResult}
+import pl.touk.nussknacker.engine.api.{Context, LazyParameter}
+
+/**
+  * Purpose of this trait is to hide the fact, that the real implementation of LazyParameter can has postponed
+  * evaluator's creation.
+  */
+trait LazyParameterWithPotentiallyPostponedEvaluator[+T <: AnyRef] extends LazyParameter[T] {
```

Comment:
We can experiment in the proposed direction but if this change has influence on the Flink component's API (especially `CustomStreamTransformer`), then let's discuss it at the end, the best in the separate PR. I mention about that, because I feel that making it cleaner in the `interpreter` module, may introduce a drawback for a developer of an extension, that he/she has concepts that look differently depending on the engine. I'm not saying no, but just let's talk about that in the separate PR and consider all props and cons.

Reply:
done